src/cache.rs

```diff
@@ -217,14 +220,12 @@ impl Cache {
             .collect();
 
         // Look up custom page (<name>.page). If it exists, return it directly
-        if let Some(config_dir) = custom_pages_dir {
-            let custom_page = config_dir.join(custom_filename);
-            if custom_page.exists() && custom_page.is_file() {
-                return Some(PageLookupResult::with_page(custom_page));
-            }
+        let custom_page = custom_pages_dir.join(custom_filename);
+        if custom_page.exists() && custom_page.is_file() {
+            return Some(PageLookupResult::with_page(custom_page));
         }
 
-        let patch_path = Self::find_patch(&patch_filename, custom_pages_dir.as_deref());
+        let patch_path = Self::find_patch(&patch_filename, custom_pages_dir);
```

Comment:
The logic that checks for custom pages is on the same abstraction layer as `Self::find_path`, yet one is inline here and the other is it's own function. Maybe `find_patch` is not needed at all anymore if the directory has to be specified anyways. What do you think?

Reply:
(keeping in mind that we may allow not specifying this dir after all)

Reply:
Yeah, I think inlining the patch finding would be more elegant now, especially using `Option::filter` to make it a 1 liner